ui/ducks/bridge/actions.ts

```diff
@@ -86,3 +118,330 @@ export const updateQuoteRequestParams = (params: Partial<QuoteRequest>) => {
     );
   };
 };
+
+export const getBridgeERC20Allowance = async (
+  contractAddress: string,
+  chainId: Hex,
+): Promise<string> => {
+  return await submitRequestToBackground(
+    BridgeBackgroundAction.GET_BRIDGE_ERC20_ALLOWANCE,
+    [contractAddress, chainId],
+  );
+};
+
+export const submitBridgeTransaction = (
+  quoteResponse: QuoteResponse,
+  history: ReturnType<typeof useHistory>,
+) => {
+  return async (
+    dispatch: MetaMaskReduxDispatch,
+    getState: () => MetaMaskReduxState & BridgeAppState,
+  ) => {
+    const state = getState();
+    // Track event TODO
+
+    const calcFeePerGas = () => {
```

Comment:
Rather than embedding lots of nested inline functions, would it be cleaner to define these as standard utility functions at the bottom of this file?

Or alternatively, break this up into separate React hooks to support the React components?